### ISSUE TITLE
Tension between dependency injection and tandem structure

### ISSUE THREAD

**Issue by samreid** [phetsims/states-of-matter#307]
Discovered during the review in https://github.com/phetsims/states-of-matter/issues/299, I noticed 8 occurrences of the string literal `Tandem = `, like so:

```js
  // create the root tandem for the node that includes the atom diameter title and slider
  const atomDiameterSliderTandem = options.tandem.createTandem( 'atomDiameterSlider' );
```

Which is then used like so:

```js
  // add atom diameter slider
  const atomDiameterTitle = new Text( atomDiameterString, {
    font: NORMAL_TEXT_FONT,
    fill: options.panelTextFill,
    maxWidth: SLIDER_TITLE_MAX_WIDTH,
    tandem: atomDiameterSliderTandem.createTandem( 'title' )
  } );

// ...

  const atomDiameterSlider = new HSlider(
    dualAtomModel.adjustableAtomDiameterProperty,
    new Range( SOMConstants.MIN_SIGMA, SOMConstants.MAX_SIGMA ),
    merge( { tandem: atomDiameterSliderTandem.createTandem( 'slider' ) }, commonSliderOptions )
  );
```

In https://github.com/phetsims/phet-io/issues/1617#issuecomment-624984992 it was recommended to try a subclassing pattern where appropriate.  Would that work for these 8 cases?

**Comment by samreid** [comment]
I'm seeing something similar in this line in js/atomic-interactions/view/InteractivePotentialGraph.js

```js
addEpsilonDragListener( this.epsilonControls.line, this.epsilonControls.line.tandem.createTandem( 'dragListener' ) );
```

**Comment by samreid** [comment]
I'm seeing something similar in this line in js/atomic-interactions/view/InteractivePotentialGraph.js

```js
addEpsilonDragListener( this.epsilonControls.line, this.epsilonControls.line.tandem.createTandem( 'dragListener' ) );
```

**Comment by jbphet** [comment]
@samreid pointed out that there were 8 occurrences of `*Tandem =`.  I've addressed all but two.  For the ones that I *did* address, I did some subclassing some cases and created and used the tandem from the created object in others.  All in all, it feels like an improvement.

The two that I did *not* address were for radio button groups, and I wasn't sure how to improve these.  Radio buttons create their sub-components based on descriptors that are provided, so we either need to do what I've done and change `RadioButtonGroup` to automatically create sub-tandems, which seems to be potentially challenging.

@samreid - back to you to review the 6 that have been addressed and to either advise me on the other two, approve it as is, or log an issue about how to make this work better for radio button groups.
